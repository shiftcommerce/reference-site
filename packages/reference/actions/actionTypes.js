// Action types
export const SET_PRODUCT = 'SET_PRODUCT'
export const UPDATE_CART_LINE_ITEMS = 'UPDATE_CART_LINE_ITEMS'
export const STORE_CART = 'STORE_CART'
export const SET_SEARCH_STATE = 'SET_SEARCH_STATE'
export const SET_SEARCH_QUERY = 'SET_SEARCH_QUERY'
export const SET_CATEGORY = 'SET_CATEGORY'
export const SET_CATEGORIES = 'SET_CATEGORIES'
export const SET_CHECKOUT_INPUT_VALUE = 'SET_CHECKOUT_INPUT_VALUE'
export const SET_CHECKOUT_INPUT_VALIDATION_MESSAGE = 'SET_CHECKOUT_INPUT_VALIDATION_MESSAGE'
export const STORE_CHECKOUT = 'STORE_CHECKOUT'
export const SET_CHECKOUT_INPUT_SHOWN = 'SET_CHECKOUT_INPUT_SHOWN'
export const SET_SHIPPING_METHOD = 'SET_SHIPPING_METHOD'
export const TOGGLE_CHECKOUT_COMPONENT_COLLAPSED = 'TOGGLE_CHECKOUT_COMPONENT_COLLAPSED'
export const CREATE_ORDER = 'CREATE_ORDER'
export const SET_SHIPPING_AS_BILLING_KEY = 'SET_SHIPPING_AS_BILLING_KEY'
export const INITIATE_CHECKOUT = 'INITIATE_CHECKOUT'
export const INITIATE_CART = 'INITIATE_CART'
export const CHANGE_PAYMENT_METHOD = 'CHANGE_PAYMENT_METHOD'
export const REQUEST_CARD_TOKEN = 'REQUEST_CARD_TOKEN'
export const SET_CARD_TOKEN = 'SET_CARD_TOKEN'
export const SET_PAYMENT_ERROR = 'SET_PAYMENT_ERROR'
