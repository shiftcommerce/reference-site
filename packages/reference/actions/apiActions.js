// Libs
import ApiClient from '../lib/ApiClient'
import JsonApiParser from '../lib/JsonApiParser'

export const readEndpoint = (request) => {
  return (dispatch, getState) => {
    dispatch(fetching(request.requestActionType))
    return new ApiClient().read(request.endpoint, request.query)
      .then(response => {
        if (response.status === 200 || response.status === 304) {
          const parsedPayload = new JsonApiParser().parse(response.data)
          dispatch(sendResponse(request.successActionType, parsedPayload))
        } else {
          dispatch(setErroredTo(request.errorActionType, response.data))
        }
      })
      .catch((error) => {
        dispatch(setErroredTo(request.errorActionType, error))
      })
  }
}

export const postEndpoint = (request) => {
  return (dispatch, getState) => {
    dispatch(fetching(request.requestActionType))
    return new ApiClient().post(request.endpoint, request.body)
      .then(response => {
        _determinePostDispatch(dispatch, request, response)
      })
      .catch(error => {
        dispatch(setErroredTo(request.errorActionType, error))
      })
  }
}

export function sendResponse (actionType, parsedPayload) {
  return {
    type: actionType,
    payload: parsedPayload
  }
}

function fetching (actionType) {
  return {
    type: actionType
  }
}

function setErroredTo (actionType, data) {
  return {
    type: actionType,
    payload: {
      error: {
        data: data
      }
    }
  }
}

function _checkForErrors (response) {
  let validationPassed = true
  // Validation for customer account form
  if (response.data.length > 0) {
    response.data.forEach(item => {
      if (parseInt(item.status) === 422 || parseInt(item.status) === 404) {
        validationPassed = false
      }
    })
  }

  return validationPassed
}

function _determinePostDispatch (dispatch, request, response) {
  const validationPassed = _checkForErrors(response)

  if (response.status === 201 || response.status === 200) {
    if (validationPassed) {
      const parsedPayload = new JsonApiParser().parse(response.data)
      dispatch(sendResponse(request.successActionType, parsedPayload))
    } else {
      dispatch(setErroredTo(request.errorActionType, response.data))
    }
  } else {
    dispatch(setErroredTo(request.errorActionType, response.data))
  }
}
