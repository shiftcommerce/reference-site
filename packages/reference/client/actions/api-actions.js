// Libs
import ApiClient from '../lib/api-client'
import JsonApiParser from '../lib/json-api-parser'

export const readEndpoint = (request) => {
  return (dispatch) => {
    dispatchInitialAction(dispatch, request)
    return new ApiClient().read(request.endpoint, request.query)
      .then(processResponse(dispatch, request, [200, 304]))
      .catch((error) => {
        if (request.errorActionType) dispatch(setErroredTo(request.errorActionType, error, request))
      })
  }
}

export const postEndpoint = (request) => {
  return (dispatch) => {
    dispatchInitialAction(dispatch, request)
    return new ApiClient().post(request.endpoint, request.body)
      .then(response => {
        _determinePostDispatch(dispatch, request, response)
      })
      .catch(error => {
        if (request.errorActionType) dispatch(setErroredTo(request.errorActionType, error))
      })
  }
}

export const deleteEndpoint = (request) => {
  return (dispatch) => {
    dispatchInitialAction(dispatch, request)
    return new ApiClient().delete(request.endpoint, request.body)
      .then(processResponse(dispatch, request, [204]))
      .catch(error => {
        if (request.errorActionType) dispatch(setErroredTo(request.errorActionType, error))
      })
  }
}

function dispatchInitialAction (dispatch, request) {
  if (request.requestActionType) {
    dispatch(fetching(request.requestActionType, request.requestActionData))
  }
}

function processResponse (dispatch, request, expectedStatusCodes) {
  return (response) => {
    if (expectedStatusCodes.includes(response.status)) {
      if (request.successActionType) {
        const parsedPayload = new JsonApiParser().parse(response.data)
        dispatch(sendResponse(request.successActionType, parsedPayload))
      }
    } else {
      if (request.errorActionType) dispatch(setErroredTo(request.errorActionType, response.data, request))
    }
  }
}

export function sendResponse (actionType, parsedPayload) {
  return {
    type: actionType,
    payload: parsedPayload
  }
}

function fetching (actionType, actionData) {
  return {
    type: actionType,
    data: actionData
  }
}

function setErroredTo (actionType, data, request) {
  return {
    type: actionType,
    payload: {
      error: {
        data: data,
        request: request
      }
    }
  }
}

function _checkForErrors (response) {
  let validationPassed = true
  // Validation for customer account form
  if (response.data.length > 0) {
    response.data.forEach(item => {
      if (parseInt(item.status) === 422 || parseInt(item.status) === 404) {
        validationPassed = false
      }
    })
  }

  return validationPassed
}

function _determinePostDispatch (dispatch, request, response) {
  const validationPassed = _checkForErrors(response)

  if ((response.status === 202 || response.status === 201 || response.status === 200) && validationPassed) {
    if (request.successActionType) {
      const parsedPayload = new JsonApiParser().parse(response.data)
      dispatch(sendResponse(request.successActionType, parsedPayload))
    }
  } else {
    dispatch(setErroredTo(request.errorActionType, response.data))
  }
}
