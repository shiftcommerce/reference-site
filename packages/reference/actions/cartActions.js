// actionTypes
import * as actionTypes from './actionTypes'

// cartHandler
import * as cartHandler from './handlers/cartHandler'

// initial states
import { cartInitialState } from './../reducers/setCart'
import { checkoutInitialState } from './../reducers/setCheckout'

// localForage
import LocalForage from '../lib/localforage'
const localForage = new LocalForage()

// Return to reducer with the updated cart details
export function updateCart (cart) {
  return {
    type: actionTypes.UPDATE_CART_LINE_ITEMS,
    cart: Object.assign({}, cart, {loading: false})
  }
}

// Store the cart details in to the local redux store
export function storeCart (cart) {
  return {
    type: actionTypes.STORE_CART,
    cart: Object.assign({}, cart, {loading: false})
  }
}

export function initiateCart () {
  return {
    type: actionTypes.INITIATE_CART
  }
}

export function initiateCheckout () {
  return {
    type: actionTypes.INITIATE_CHECKOUT
  }
}

// This function will be called by minibag on every page load
// It will initialize cart lineitems
export function readCart () {
  return (dispatch, getState) => {
    let cart = getState().cart

    // check if local redux has line items
    if (cart.totalQuantity === 0) {
      // check if indexedDB, if local redux store dont have line item
      // This is to ensure, we dont miss line items on page refresh
      return localForage.getValue('cart').then((cart) => {
        // if indexedDB also dont have lineitems,
        // initalize cart
        if (cart === null) {
          dispatch(initializeCart())
        }
        // update local redux store
        dispatch(storeCart(cart))

        return cart
      })
    // if local redux store has line items,
    // dispatch them
    } else {
      dispatch(storeCart(cart))
      return Promise.resolve(cart)
    }
  }
}

// This function will be called, when user add items to bag
// in product detail page. If a line item already exists, it
// will add the new quantity to the existing one
export function addToCart (lineItem) {
  return (dispatch, getState) => {
    let cart = getState().cart
    addOrUpdateLineItems(cart, lineItem, dispatch)
  }
}

export function updateQuantity (lineItem) {
  return (dispatch, getState) => {
    let cart = getState().cart
    cart = cartHandler.updateCart(cart, lineItem)
    dispatchUpdate(cart, dispatch)
  }
}

export function initializeCart () {
  return (dispatch) => {
    dispatch(initiateCart())
    dispatch(initiateCheckout())
    localForage.setValue('cart', cartInitialState)
    localForage.setValue('checkout', checkoutInitialState)
  }
}

// Private functions

function addOrUpdateLineItems (cart, lineItem, dispatch) {
  cart = cartHandler.addToCart(cart, lineItem)
  dispatchUpdate(cart, dispatch)
}

function dispatchUpdate (cart, dispatch) {
  // update local redux store
  dispatch(updateCart(cart))
  // update indexedDB
  localForage.setValue('cart', cart)
}
