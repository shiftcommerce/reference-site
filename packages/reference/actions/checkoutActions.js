// actionTypes
import * as actionTypes from './actionTypes'

// checkout Handler
import * as checkoutHandler from './handlers/checkoutHandler'

// localForage
import LocalForage from '../lib/localforage'
const localForage = new LocalForage()

export function inputChange (formName, fieldName, fieldValue) {
  return (dispatch) => {
    dispatch(storeInputChange(formName, fieldName, fieldValue))
  }
}
// Store the input change info in the local redux store
export function storeInputChange (formName, fieldName, fieldValue) {
  return {
    type: actionTypes.SET_CHECKOUT_INPUT_VALUE,
    payload: {
      formName: formName,
      fieldName: fieldName,
      fieldValue: fieldValue
    }
  }
}

export function showField (formName, fieldName) {
  return {
    type: actionTypes.SET_CHECKOUT_INPUT_SHOWN,
    payload: {
      formName: formName,
      fieldName: fieldName
    }
  }
}

export function toggleCollapsed (componentName) {
  return {
    type: actionTypes.TOGGLE_CHECKOUT_COMPONENT_COLLAPSED,
    payload: {
      componentName: componentName
    }
  }
}

export function inputComplete () {
  return (dispatch, getState) => {
    const checkout = getState().checkout
    storeCheckoutInLocalStorage(checkout)
  }
}

// Stores a checkout in redux store
export function storeCheckout (checkout) {
  return {
    type: actionTypes.STORE_CHECKOUT,
    checkout: checkout
  }
}

// set shipping method details
export function setShippingMethod (shippingMethod) {
  return (dispatch, getState) => {
    const checkout = checkoutHandler.updateShippingMethod(getState().checkout, shippingMethod)
    dispatch(storeShippingMethod(checkout))
    storeCheckoutInLocalStorage(checkout)
  }
}

// Store shipping method details
export function storeShippingMethod (checkout) {
  return {
    payload: checkout,
    type: actionTypes.SET_SHIPPING_METHOD
  }
}

// Stores checkout in indexedDB, eventually we can update and/or duplicate this for API?
export function storeCheckoutInLocalStorage (checkout) {
  localForage.setValue('checkout', checkout)
}

// This function will be called by checkout every time the component is mounted
export function readCheckoutFromLocalStorage () {
  return (dispatch, getState) => {
    let checkout = getState().checkout
    // check if local redux store has no data
    if (checkout.updatedAt === undefined) {
      localForage.getValue('checkout').then((checkout) => {
        // if indexedDB is also empty, initialize an empty checkout
        if (checkout === null) {
          checkout = {
            shippingAddress: {},
            billingAddress: {},
            paymentMethod: {},
            shippingAddressAsBillingAddress: true,
            shippingMethod: {},
            currentStep: 1
          }
        }
        // update local redux store
        dispatch(storeCheckout(checkout))
      })
    // if local redux store has checkout data, dispatch it
    } else {
      dispatch(storeCheckout(checkout))
    }
    return checkout
  }
}

export function setShippingBillingAddress (formName, fieldName, fieldValue) {
  return (dispatch, getState) => {
    dispatch(setBillingAsShippingKey(formName, fieldName, fieldValue))
  }
}

function setBillingAsShippingKey (formName, fieldName, fieldValue) {
  return {
    type: actionTypes.SET_SHIPPING_AS_BILLING_KEY,
    payload: {
      formName: formName,
      fieldName: fieldName,
      fieldValue: fieldValue
    }
  }
}

export function changePaymentMethod (paymentMethod) {
  return {
    type: actionTypes.CHANGE_PAYMENT_METHOD,
    paymentMethod: paymentMethod
  }
}
