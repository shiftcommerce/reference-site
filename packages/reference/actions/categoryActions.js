// actionTypes
import * as types from './actionTypes'

// Actions
import { readEndpoint } from './apiActions'

export function readCategories () {
  const request = {
    endpoint: '/getCategories',
    query: {
      fields: {
        categories: 'id,title,reference'
      }
    },
    successActionType: types.SET_CATEGORIES
  }
  return readEndpoint(request)
}

export function readCategory (categoryID) {
  const request = {
    endpoint: `/getCategory/${categoryID}`,
    query: {
      fields: {
        categories: 'id,title,reference'
      }
    },
    successActionType: types.SET_CATEGORY
  }
  return readEndpoint(request)
}
