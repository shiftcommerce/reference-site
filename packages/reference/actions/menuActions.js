// actionTypes
import * as types from './actionTypes'

// Config
import { menuRequest } from '../requests/menuRequest'

// Actions
import { readEndpoint } from './apiActions'

export function readMenu (store) {
  const menu = store.getState().menu.data
  if (menu.length === 0) {
    return readEndpoint(menuRequest, menuRequest.successActionType)
  } else {
    return {
      type: types.SET_MENU,
      payload: {
        data: menu
      }
    }
  }
}
