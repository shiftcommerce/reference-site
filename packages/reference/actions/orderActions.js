// actionTypes
import * as types from './actionTypes'

// Actions
import { postEndpoint } from './apiActions'

// Handler
import { convertCheckoutToOrder } from './handlers/orderHandler'

export function createOrder (cart, checkout) {
  const request = {
    endpoint: `/createOrder`,
    body: convertCheckoutToOrder(cart, checkout),
    successActionType: types.CREATE_ORDER
  }
  return postEndpoint(request)
}
