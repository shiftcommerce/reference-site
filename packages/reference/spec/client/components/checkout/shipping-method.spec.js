// Libraries
import nock from 'nock'

// Component
import ShippingMethod from '../../../../client/components/checkout/shipping-methods'
import * as apiActions from '../../../../client/actions/api-actions'

afterEach(() => { nock.cleanAll() })

test('render shipping methods as expected', async () => {
  // Arrange
  const cart = {
    line_items: [
      {
        id: '1'
      }
    ],
    line_items_count: 1,
    shipping_method: {}
  }

  const checkout = {
    shippingAddress: {
      collapsed: true,
      completed: true
    },
    shippingMethod: {}
  }

  const fetchShippingSpy = jest.spyOn(ShippingMethod, 'fetchShippingMethods').mockImplementation(() => Promise.resolve({
    data: [{
      id: 1,
      sku: 'STA_SHIP',
      label: 'Standard shipping',
      total: 3.75,
      meta_attributes: {
        working_days: {
          value: 2
        }
      }
    }]
  }))

  // Act
  const wrapper = mount(<ShippingMethod cart={cart} checkout={checkout} dispatch={jest.fn()} />)

  // Assert
  expect(wrapper).toIncludeText('Loading...')
  expect(wrapper).toMatchSnapshot()

  await wrapper.instance().componentDidMount()

  expect(wrapper).toIncludeText('1 item')
  expect(wrapper).toIncludeText('Standard shipping')

  fetchShippingSpy.mockRestore()
})

test('renders line item quantity as expected', async () => {
  // Arrange
  const cart = {
    line_items: [],
    line_items_count: 0,
    shipping_method: {
      id: 1,
      sku: 'STA_SHIP',
      label: 'Standard shipping',
      total: 3.75,
      meta_attributes: {
        working_days: {
          value: 2
        }
      }
    }
  }

  const checkout = {
    shippingAddress: {
      collapsed: true,
      completed: true
    },
    shippingMethod: {
      collapsed: false
    }
  }

  const fetchShippingSpy = jest.spyOn(ShippingMethod, 'fetchShippingMethods').mockImplementation(() => Promise.resolve({ data: [] }))

  // Act
  const wrapper = mount(<ShippingMethod cart={cart} checkout={checkout} />)

  // Assert
  expect(wrapper).toIncludeText('Loading...')

  await wrapper.instance().componentDidMount()

  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('0 items')

  fetchShippingSpy.mockRestore()
})

test('preselects first shipping method when fetching shipping methods and none is set', async () => {
  // Arrange
  const cart = {
    line_items: [],
    line_items_count: 0
  }

  const checkout = {
    shippingAddress: {
      collapsed: true,
      completed: true
    },
    shippingMethod: {
      collapsed: false
    }
  }

  const fetchShippingSpy = jest.spyOn(ShippingMethod, 'fetchShippingMethods').mockImplementation(() => Promise.resolve({
    data: [{
      id: 1,
      sku: 'STA_SHIP',
      label: 'Standard shipping',
      total: 3.75,
      meta_attributes: {
        working_days: {
          value: 2
        }
      }
    }, {
      id: 2,
      sku: 'NEX_SHIP',
      label: 'Next day delivery',
      total: 6.25,
      meta_attributes: {
        working_days: {
          value: 1
        }
      }
    }]
  }))

  const postSpy = jest.spyOn(apiActions, 'postEndpoint').mockImplementation(() => { cart.shipping_method = { id: 1 } })

  // Act
  const wrapper = mount(<ShippingMethod cart={cart} checkout={checkout} dispatch={jest.fn()} />)

  // Assert
  expect(wrapper).toIncludeText('Loading...')

  await wrapper.instance().componentDidMount()

  const request = postSpy.mock.calls[0][0]
  expect(request.endpoint).toEqual('/setShippingMethod')
  expect(request.body.shippingMethodId).toEqual(1)

  postSpy.mockRestore()
  fetchShippingSpy.mockRestore()
})

test('selecting a shipping method makes a correct API call', async () => {
  // Arrange
  const cart = {
    line_items: [],
    line_items_count: 1,
    shipping_method: {
      id: 1,
      sku: 'STA_SHIP',
      label: 'Standard shipping',
      total: 3.75,
      meta_attributes: {
        working_days: {
          value: 2
        }
      }
    }
  }

  const checkout = {
    shippingAddress: {
      collapsed: true,
      completed: true
    },
    shippingMethod: {
      collapsed: false
    }
  }

  const fetchShippingSpy = jest.spyOn(ShippingMethod, 'fetchShippingMethods').mockImplementation(() => Promise.resolve({
    data: [{
      id: 1,
      sku: 'STA_SHIP',
      label: 'Standard shipping',
      total: 3.75,
      meta_attributes: {
        working_days: {
          value: 2
        }
      }
    }, {
      id: 2,
      sku: 'NEX_SHIP',
      label: 'Next day delivery',
      total: 6.25,
      meta_attributes: {
        working_days: {
          value: 1
        }
      }
    }]
  }))

  const postSpy = jest.spyOn(apiActions, 'postEndpoint').mockImplementation(() => { cart.shipping_method = { id: 1 } })

  // Act
  const wrapper = mount(<ShippingMethod cart={cart} checkout={checkout} dispatch={jest.fn()} />)

  // Assert
  expect(wrapper).toIncludeText('Loading...')

  await wrapper.instance().componentDidMount()
  wrapper.update()

  wrapper.find('input[id="NEX_SHIP_2"]').simulate('change', { target: { checked: true } })

  const request = postSpy.mock.calls[0][0]
  expect(request.endpoint).toEqual('/setShippingMethod')
  expect(request.body.shippingMethodId).toEqual(2)

  fetchShippingSpy.mockRestore()
  postSpy.mockRestore()
})

test('fetchShippingMethods() returns shipping methods from the API', async () => {
  nock(process.env.API_HOST_PROXY)
    .defaultReplyHeaders({ 'access-control-allow-origin': '*' })
    .get('/getShippingMethods')
    .reply(200, { data: 'shipping methods data' })

  expect(await ShippingMethod.fetchShippingMethods()).toEqual({ data: 'shipping methods data' })
})

test('fetches shipping methods, sorts them by total and puts them in state', async () => {
  const shippingMethods = {
    data: [{
      id: 1,
      total: 20,
      meta_attributes: {
        working_days: {
          value: 1
        }
      }
    }, {
      id: 2,
      total: 10,
      meta_attributes: {
        working_days: {
          value: 1
        }
      }
    }]
  }

  const cart = {
    shipping_method: {}
  }

  const instance = shallow(<ShippingMethod cart={cart} />).instance()

  const fetchShippingMethodsSpy = jest.spyOn(instance.constructor, 'fetchShippingMethods').mockImplementation(() => shippingMethods)

  await instance.componentDidMount()

  expect(instance.state).toEqual({
    loading: false,
    shippingMethods: shippingMethods.data
  })

  fetchShippingMethodsSpy.mockRestore()
})
