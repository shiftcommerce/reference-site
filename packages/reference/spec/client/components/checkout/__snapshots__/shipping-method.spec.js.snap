// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render collapsed view as expected 1`] = `
<ShippingMethods
  cart={
    Object {
      "line_items": Array [
        Object {
          "id": "1",
        },
      ],
      "line_items_count": 1,
      "shipping_method": Object {
        "id": 1,
        "label": "Standard shipping",
        "meta_attributes": Object {
          "working_days": Object {
            "value": 2,
          },
        },
        "sku": "STA_SHIP",
        "total": 3.75,
      },
    }
  }
  checkout={
    Object {
      "shippingAddress": Object {
        "collapsed": true,
        "completed": true,
      },
      "shippingMethod": Object {
        "collapsed": true,
      },
    }
  }
  dispatch={[MockFunction]}
>
  <div
    aria-label="Shipping Methods"
    className="o-form c-shipping-method"
  >
    <p>
      Loading...
    </p>
  </div>
</ShippingMethods>
`;

exports[`render shipping methods as expected 1`] = `
<ShippingMethods
  cart={
    Object {
      "line_items": Array [
        Object {
          "id": "1",
        },
      ],
      "line_items_count": 1,
      "shipping_method": Object {},
    }
  }
  checkout={
    Object {
      "shippingAddress": Object {
        "collapsed": true,
        "completed": true,
      },
      "shippingMethod": Object {},
    }
  }
  dispatch={[MockFunction]}
>
  <div
    aria-label="Shipping Methods"
    className="o-form c-shipping-method"
  >
    <p>
      Loading...
    </p>
  </div>
</ShippingMethods>
`;

exports[`renders line item quantity as expected 1`] = `
<ShippingMethods
  cart={
    Object {
      "line_items": Array [],
      "line_items_count": 0,
      "shipping_method": Object {
        "id": 1,
        "label": "Standard shipping",
        "meta_attributes": Object {
          "working_days": Object {
            "value": 2,
          },
        },
        "sku": "STA_SHIP",
        "total": 3.75,
      },
    }
  }
  checkout={
    Object {
      "shippingAddress": Object {
        "collapsed": true,
        "completed": true,
      },
      "shippingMethod": Object {
        "collapsed": false,
      },
    }
  }
>
  <div
    aria-label="Shipping Methods"
    className="o-form c-shipping-method"
  >
    <p>
      Loading...
    </p>
  </div>
</ShippingMethods>
`;
